forc-pkg/src/source/reg/mod.rs

```diff
@@ -479,6 +493,58 @@ async fn fetch(fetch_id: u64, pinned: &Pinned, ipfs_node: &IPFSNode) -> anyhow::
     Ok(path)
 }
 
+/// Fetches package from CDN as a fallback when IPFS fails
+async fn fetch_from_s3(pinned: &Pinned, path: &Path) -> anyhow::Result<()> {
+    let client = reqwest::Client::new();
+
+    // Construct CDN URL directly from IPFS hash
+    let cdn_url = format!("http://cdn.forc.pub/{}", pinned.cid.0);
+
+    println_action_green("Fetching", "from forc.pub");
```

Comment:
maybe we can improve this message, when trying from IPFS it prints 

```  
Fetching from http://ipfs.forc.pub. Note: This can take several minutes.
````

Perhaps for consistency we could print. 

```
Fetching from http://cdn.forc.pub. Note: This can take several minutes.
```

Reply:
Oh yeah makes sense

Reply:
done in f2ab249